src/main/context/AuthContext.jsx

```diff
@@ -0,0 +1,48 @@
+import { createContext, useEffect, useState } from "react";
+
+import { GoogleAuthProvider, signInWithPopup, signOut, onAuthStateChanged } from "firebase/auth";
+import { auth } from "../firebase";
+
+export const AuthContext = createContext()
+
+export const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState()
+  const [firebaseUser, setFirebaseUser] = useState();
+
+
+  // google signin
+  const googleSignIn = () => {
+    const provider = new GoogleAuthProvider();
+    signInWithPopup(auth, provider)
+  }
+
+  // signout
+  const logOut = () => {
+    signOut(auth)
```

Comment:
clear currentUser nad user values from state after login and navigate to login page

Reply:
cleared user values and added navigate to signin 

need help with restricting user from entering the dashboard page without login.

Reply:
refer Protected Routes